src/main/java/com/example/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.UserRegistrationRequestDto;
+import com.example.onlinebookstore.dto.UserResponseDto;
+import com.example.onlinebookstore.exception.RegisterationException;
+import com.example.onlinebookstore.mapper.UserMapper;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.user.UserRepository;
+import com.example.onlinebookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto) {
+        if (userRepository.findByEmail(requestDto.getEmail()) != null) {
```

Comment:
It's better to create another method in user repo
```suggestion
        if (userRepo.existsByEmail(request.email())) {
```